include/RAJA/policy/cuda/kernel/For.hpp

```diff
@@ -119,19 +117,28 @@ struct CudaStatementExecutor<
 
 
   static
-  inline RAJA_DEVICE void exec(Data &data)
+  inline RAJA_DEVICE void exec(Data &data, bool thread_active)
   {
     // block stride loop
     int len = segment_length<ArgumentId>(data);
     auto i0 = get_cuda_dim<ThreadDim>(threadIdx);
     auto i_stride = get_cuda_dim<ThreadDim>(blockDim);
-    for(auto i = i0;i < len;i += i_stride){
+    auto i = i0;
+    for(;i < len;i += i_stride){
```

Comment:
is there a reason to initialize the loop variable outside of the for statement?

Reply:
Adam just misses C so much.

Reply:
below it has to use the final value of i0 to determine if thread masking needs to occur

Reply:
OK. Fine.